modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rel/IgniteFilter.java

```diff
@@ -171,6 +171,16 @@ public IgniteFilter(RelInput input) {
                 // Remap correlate fields to input fields.
                 int corrFieldsCnt = ((RexNode)corr.getNode()).getType().getFieldCount();
                 int inputFieldsCnt = getRowType().getFieldCount();
+
+                // Calcite during planning can add fields to the left hand of the correlated join, but without proper
```

Comment:
Minor: 'add fields' mihght be changed to smth. like 'swap' or 'commute'. Actually, the test query's join has the same 4 fields of the row type. This is result of JoinCommuteRule. It seems not to produce other, wider join row types.